cli/command/image/build.go

```diff
@@ -94,6 +94,15 @@ func NewBuildCommand(dockerCli *command.DockerCli) *cobra.Command {
 		Args:  cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			options.context = args[0]
+
+			flags := cmd.Flags()
+			if !flags.Changed("stream") && isSessionSupported(dockerCli) {
+				// if stream is not set try the default from config file
+				if v := dockerCli.ConfigFile().BuildStreamContext; v != nil {
+					options.stream = *v
+				}
+			}
```

Comment:
You mean `setDefaultsFromConfig(dockerCli command.Cli, flags *pflag.FlatSet, options *buildOptions)` ? Need a pointer of the options to change it. What is the bool return supposed to mean? Also, this wouldn't work because this check needs `DockerCli` not `Cli` to check if session is supported.